backend/onedrive/onedrive.go

```diff
@@ -374,6 +388,12 @@ func Config(ctx context.Context, name string, m configmap.Mapper, config fs.Conf
 	region, graphURL := getRegionURL(m)
 
 	if config.State == "" {
+		disableSitePermission, ok := m.Get("disable_site_permission")
+		if disableSitePermission == "true" && ok {
+			oauthConfig.Scopes = scopesWithoutSitePermission
+		} else {
```

Comment:
We should be able to get rid of this else branch. (The same for the other if-else block)

Reply:
How? Should I lift the `oauthConfig.Scopes = scopesWithSitePermission` up before the `if` statement? The `scopesWithSitePermission` is configured as the default, so here it's just for belt and braces..

Reply:
I initially thought we could remove it altogether, but now I see that the object can get reused, so I think it's good as it is, no changes needed.